### PR TITLE
Linux VM custom_data using templatefile

### PR DESCRIPTION
**Pull request by jvdhaterd**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/ISSUE-ID-GOES-HERE)

## PR Checklist

---

- [x] I have added example(s) inside the [./examples/] folder
- [x] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

When building a solution for a Fortigate Firewall NVA I came across the following example [108-single-linux-storage-connect-custom-data](examples/compute/virtual_machine/108-single-linux-storage-connect-custom-data). In this example a palo_alto_connection_string is constructed using storage accounts data. As I have similar requirements for the Fortigate I created a feature to support custom_data using a [local_sensitive_file](https://registry.terraform.io/providers/hashicorp/local/latest/docs/resources/sensitive_file) resource and a [templatefile](https://developer.hashicorp.com/terraform/language/functions/templatefile) function. As this is a more generic solution, it also supports the palo_alto_connection_string use case. For backwards compatibility the original palo_alto_connection_string is preserved. 

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Testing

Use the example I provided

### CODE REVIEW

**Comment by arnaudlh** [comment]
Thanks for the PR @jvdhaterd! 
Trying to run the example you added but getting the following error. does it ring a bell?: 

```
╷
│ Error: Key "disk-key" was not found in Key Vault at URI "https://exio-kv-vmlinuxkv1.vault.azure.net/"
│ 
│   with module.example.module.virtual_machines["example_vm1"].data.azurerm_key_vault_key.custom_data["key1"],
│   on ../modules/compute/virtual_machine/dynamic_custom_data.tf line 10, in data "azurerm_key_vault_key" "custom_data":
│   10: data "azurerm_key_vault_key" "custom_data" {
│ 
╵
```

**Comment by jvdhaterd** [comment]
I have tested it and after a few tries I encountered the same error. I seems that sometimes terraform notifies the disk-key as created but the datasource can't find it yet. Running the apply a second time results in a successful apply.

**Comment by jvdhaterd** [comment]
Any suggestions?

**Comment by arnaudlh** [comment]
> Any suggestions @arnaudlh?

any depends_on we could add without adverse effects?

**Comment by jvdhaterd** [comment]
@arnaudlh I added a depends_on, issue should be resolved

**Review by arnaudlh — APPROVED** [review]
LGTM

File: examples/compute/virtual_machine/113-single-linux-custom-data-template/configuration.tfvars [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+    virtual_machine_settings = {
+      linux = {
+        name                            = "example_vm1"
+        size                            = "Standard_F2"
+        admin_username                  = "adminuser"
+        disable_password_authentication = true
+
+        custom_data = {
+          templatefile        = "custom_data.tpl"
```
Comment (arnaudlh) [inline examples/compute/virtual_machine/113-single-linux-custom-data-template/configuration.tfvars:66]:
```suggestion
          templatefile        = "compute/virtual_machine/113-single-linux-custom-data-template/custom_data.tpl"
```

**Review by jvdhaterdwestland — APPROVED** [review]

**Pull request merged**